specification/synapse/data-plane/Microsoft.Synapse/preview/2021-06-01-preview/runNotebook.json

```diff
@@ -0,0 +1,570 @@
+          "x-nullable": true,
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "kind": {
+          "x-nullable": true,
+          "type": "string"
+        },
+        "proxyUser": {
+          "x-nullable": true,
+          "type": "string"
+        },
+        "artifactId": {
+          "type": "string"
+        },
+        "name": {
+          "x-nullable": true,
+          "type": "string"
+        },
+        "file": {
+          "x-nullable": true,
+          "type": "string"
+        },
+        "className": {
+          "x-nullable": true,
+          "type": "string"
+        },
+        "args": {
+          "x-ms-client-name": "arguments",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "jars": {
+          "uniqueItems": false,
+          "x-nullable": true,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "pyFiles": {
+          "x-ms-client-name": "pythonFiles",
+          "uniqueItems": false,
+          "x-nullable": true,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "files": {
+          "uniqueItems": false,
+          "type": "array",
+          "x-nullable": true,
+          "items": {
+            "type": "string"
+          }
+        },
+        "archives": {
+          "uniqueItems": false,
+          "x-nullable": true,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "queue": {
+          "type": "object",
+          "x-nullable": true
+        },
+        "conf": {
+          "x-ms-client-name": "configuration",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "driverMemory": {
+          "type": "string"
+        },
+        "driverCores": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "executorMemory": {
+          "type": "string"
+        },
+        "executorCores": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "numExecutors": {
+          "x-ms-client-name": "executorCount",
+          "format": "int32",
+          "type": "integer"
+        },
+        "isQueueable": {
+          "type": "boolean",
+          "x-nullable": true
+        },
+        "heartbeatTimeoutInSecond": {
+          "type": "integer",
+          "format": "int32",
+          "x-nullable": true
+        }
+      }
+    },
+    "RunNotebookResponse": {
+      "description": "Run notebook response.",
+      "type": "object",
+      "properties": {
+        "message": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Response message."
+        },
+        "result": {
+          "$ref": "#/definitions/Result"
```

Comment:
Is there some reason to nest all the result information inside this `result` property?  I don't see the `Result` schema used anywhere other than here, so it doesn't seem to be for reuse.  Flat structures are generally preferred to nested, so could you just put all those properties here?

Reply:
We can not do this as of now. Internal implantation which is used by ADF has been already in GA. It may break if customer would have started parsing the response.